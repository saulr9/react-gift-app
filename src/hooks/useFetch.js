import { useState, useEffect } from 'react';
import getGifs from '../utils/getGifs';

function useFetch(category) {
  const [state, setState] = useState({
    loading: true,
    data: []
  });
  useEffect(() => {
    getGifs(category).then((gifs) => setState({ loading: false, data: gifs }));
  }, [category]);

  return state;
}

export default useFetch;
