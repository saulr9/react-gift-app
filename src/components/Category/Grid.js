/* import { useState, useEffect } from 'react'; */
import PropTypes from 'prop-types';
import GridItem from './GridItem';
import useFetch from '../../hooks/useFetch';

function Grid({ category }) {
  const { loading, data: gifs } = useFetch(category);
  return (
    <>
      {loading && 'Loading...'}
      {gifs.map((gif) => (
        <GridItem key={gif.slug} gif={gif} />
      ))}
    </>
  );
}
Grid.propTypes = {
  category: PropTypes.string.isRequired
};
export default Grid;
