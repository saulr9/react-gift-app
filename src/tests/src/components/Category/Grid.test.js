import { shallow } from 'enzyme';
// eslint-disable-next-line import/no-unresolved
import Grid from 'components/Category/Grid';

import useFetch from '../../../../hooks/useFetch';

jest.mock('../../../../hooks/useFetch');

describe('Testing Category Grid Component', () => {
  const category = 'vegeta';
  test('Should match snapshot - Initial render with no data and loading true', () => {
    useFetch.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<Grid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('Should match snapshot - With gifs data and loading false', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://localhost/cualquier/cosa.jpg',
        title: 'Cualquier cosa'
      },
      {
        id: '123',
        url: 'https://localhost/cualquier/cosa.jpg',
        title: 'Otra cosa'
      }
    ];

    useFetch.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<Grid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GridItem').length).toBe(gifs.length);
  });
});
