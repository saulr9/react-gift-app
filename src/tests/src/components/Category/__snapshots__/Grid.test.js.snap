// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Category Grid Component Should match snapshot - Initial render with no data and loading true 1`] = `
<Fragment>
  Loading...
</Fragment>
`;

exports[`Testing Category Grid Component Should match snapshot - With gifs data and loading false 1`] = `
<Fragment>
  <GridItem
    gif={
      Object {
        "id": "ABC",
        "title": "Cualquier cosa",
        "url": "https://localhost/cualquier/cosa.jpg",
      }
    }
  />
  <GridItem
    gif={
      Object {
        "id": "123",
        "title": "Otra cosa",
        "url": "https://localhost/cualquier/cosa.jpg",
      }
    }
  />
</Fragment>
`;
